src/api/models/lifestyle.py

```diff
@@ -5,8 +5,120 @@
 
 
 class LifestyleProfile(BaseModel):
-    bedtime: int = None
-    wakeup_time: int = None
-    like_social: int = None
-    like_clean: int = None
-    like_quite: int = None
+    sleep_time: str # choice (before midnight/after midnight/around midnight)
+    like_social: bool 
+    like_clean: int # how tidy do they need their room to be? range from 1 to 5
+    like_music: bool # do they like to have music in the room?
+    relationship_scale: int # from 1 to 5
+    aircon: str # choice (often, occasionally, not at all) 
+    interests: List[str] = [] # select from given list, let them add on if they want (others)
+    intended_pillar: str = "NA" # choice (ISTD, EPD, ESD, ASD, DAI, NA)
+    diet: str = "None" # choice (halal, no beef, kosher, others)
+    smoking: str # choice (social smoker, not smoker, active smoker)
+
+class RoommatePreference(BaseModel): # all options are set to None unless stated otherwise
+    sleep_time: str = None
+    like_social: bool = None
+    like_music: bool = None
+    interests: List[str] = None
+    intended_pillar: str = None
+    diet: str = None
+    smoking: str = None
+    preferred_roommate: str = None
+    blacklist_roommate: List[str] = None
+
+@validator("sleep_time", pre=True, always=True)
+def validate_sleep_time(cls, v):
+    if isinstance(v, str) and v.upper() in (
+        "BEFORE MIDNIGHT",
+        "AROUND MIDNIGHT",
+        "AFTER MIDNIGHT",
+    ):
+        return v
+    else:
+        return "AROUND MIDNIGHT"
+
+@validator("social_habits", pre=True, always=True)
+def validate_social_habits(cls, v):
+    if isinstance(v, bool) or v is None:
+        return v
+    else:
+        return None 
+
+@validator("room_cleanliness", pre=True, always=True)
```

Comment:
Ok, seems like you did not understand how this `@validator` works. 
Here is the documentation: https://pydantic-docs.helpmanual.io/usage/validators/